lib/components/user/nav-login-button.tsx

```diff
@@ -0,0 +1,107 @@
+import { FormattedMessage, useIntl } from 'react-intl'
+import { NavItem } from 'react-bootstrap'
+import { User } from '@auth0/auth0-react'
+import React, { HTMLAttributes } from 'react'
+import styled from 'styled-components'
+
+import { LinkContainerWithQuery } from '../form/connected-links'
+import Dropdown from '../util/dropdown'
+
+const Avatar = styled.img`
+  height: 2em;
+  margin: -15px 0;
+  width: 2em;
+`
+
+type Link = {
+  messageId: string
+  target?: string
+  url: string
+}
+
+interface Props extends HTMLAttributes<HTMLElement> {
+  links: Link[]
+  onSignInClick: () => void
+  onSignOutClick: () => void
+  profile?: User | null
+}
+
+/**
+ * This component displays the sign-in status in the nav bar.
+ * - When a user is not logged in: display 'Sign In' as a link or button.
+ * - When a user is logged in, display an 'avatar' (retrieved from the profile prop)
+ *   and a dropdown button so the user can access more options.
+ */
+const NavLoginButton = ({
+  className,
+  id,
+  links,
+  onSignInClick,
+  onSignOutClick,
+  profile,
+  style
+}: Props): JSX.Element => {
+  const intl = useIntl()
+
+  const commonProps = {
+    className,
+    id,
+    style
+  }
+
+  // If a profile is passed (a user is logged in), display avatar and drop-down menu.
+  if (profile) {
+    const displayedName = profile.nickname || profile.name
+    return (
+      <Dropdown
+        id="user-selector"
+        label={intl.formatMessage({ id: 'components.SubNav.userMenu' })}
+        name={
+          <span>
+            <Avatar
+              alt={displayedName}
```

Comment:
Probably don't need the `alt` attribute, because the entire contents from the `name` prop is overwritten by the `label` prop, which goes into the button's `aria-label`.

Reply:
Axe demanded an `alt` tag